test/it/crdvalidation/etcd/specupdate_test.go

```diff
@@ -0,0 +1,216 @@
+// SPDX-FileCopyrightText: 2025 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package etcd
+
+import (
+	"context"
+	"testing"
+
+	"github.com/gardener/etcd-druid/test/utils"
+
+	"k8s.io/apimachinery/pkg/api/resource"
+
+	. "github.com/onsi/gomega"
+)
+
+// tests for the Update Validations:
+
+// etcd.spec.storageClass is immutable
+func TestValidateSpecStorageClass(t *testing.T) {
+	skipCELTestsForOlderK8sVersions(t)
+	testNs, g := setupTestEnvironment(t)
+
+	tests := []struct {
+		name                    string
+		etcdName                string
+		initalStorageClassName  string
+		updatedStorageClassName string
+		expectErr               bool
+	}{
+		{
+			name:                    "Valid #1: Unchanged storageClass",
+			etcdName:                "etcd-valid-1",
+			initalStorageClassName:  "gardener.cloud-fast",
+			updatedStorageClassName: "gardener.cloud-fast",
+			expectErr:               false,
+		},
+		{
+			name:                    "Invalid #1: Updated storageClass",
+			etcdName:                "etcd-invalid-1",
+			initalStorageClassName:  "gardener.cloud-fast",
+			updatedStorageClassName: "default",
+			expectErr:               true,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			etcd := utils.EtcdBuilderWithoutDefaults(test.etcdName, testNs).WithReplicas(3).Build()
+			etcd.Spec.StorageClass = &test.initalStorageClassName
+
+			cl := itTestEnv.GetClient()
+			ctx := context.Background()
+			g.Expect(cl.Create(ctx, etcd)).To(Succeed())
+
+			etcd.Spec.StorageClass = &test.updatedStorageClassName
+			updateErr := cl.Update(ctx, etcd)
+			if test.expectErr {
+				g.Expect(updateErr).ToNot(BeNil())
+			} else {
+				g.Expect(updateErr).To(BeNil())
+			}
```

Comment:
You can pull this into a separate helper function called `validateEtcdUpdation`, similar to `validateEtcdCreation`. Please add both to a `helper.go` file